proxy/src/main/java/net/md_5/bungee/connection/DownstreamBridge.java

```diff
@@ -583,6 +587,24 @@ public void handle(SetCompression setCompression) throws Exception
         server.getCh().setCompressionThreshold( setCompression.getThreshold() );
     }
 
+    @Override
+    public void handle(ResourcePackRequest resourcePackRequest) throws Exception
+    {
+        ResourcePackRequestEvent requestEvent =
+            new ResourcePackRequestEvent( server, con, resourcePackRequest.getUrl(), resourcePackRequest.getHash(), resourcePackRequest.isForced(), resourcePackRequest.getPromptMessage() );
+
+        // If the event is cancelled or if two servers use the same resource pack, do not send the packet
+        if ( bungee.getPluginManager().callEvent( requestEvent ).isCancelled() || ( con.getResourcePackHash() != null && resourcePackRequest.getHash().equals( con.getRequestedResourcePackHash() ) ) )
```

Comment:
Should the event be fired if its the same?

Reply:
okay im going to change it so it doesn't call it if they're the same